Hw1Task2/Hw1Task2/BWT.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+
+Console.WriteLine("Input string: ");
+string str = Console.ReadLine();
+string transformedString = Transformation.BWT(str);
+Console.WriteLine(transformedString);
+Console.WriteLine(Transformation.BWTinverse(transformedString));
+
+// class with two public methods for burrows-wheeler transformation
+static class Transformation
+{
+    public static string BWT(string str)
+    {
+        str += '$';
+        string[] suffixes = GetSuffixes(str);
+        sort(suffixes);
+        string result = "";
+        for (int i = 0; i < suffixes.Length; i++)
+        {
+            int index = str.Length - suffixes[i].Length - 1;
+            result += str[index < 0 ? str.Length - 1 : index];
+        }
+        return result;
+    }
+
+    private static string[] GetSuffixes(string str)
+    {
+        string[] suffixes = new string[str.Length];
+        for (int i = 0; i < str.Length; i++)
+        {
+            suffixes[i] = str.Substring(i);
+        }
+        return suffixes;
+    }
+
+    private static void sort(string[] suffixes)
+    {
+        for (int index = 1; index < suffixes.Length; index++)
+        {
+            int i = index;
+            while (i > 0 && String.Compare(suffixes[i], suffixes[i-1]) < 0)
```

Comment:
```suggestion
            while (i > 0 && String.Compare(suffixes[i], suffixes[i - 1]) < 0)
```
Более того, в C# есть перегрузка операторов, поэтому `suffixes[i] < suffixes[i - 1]` тоже можно. Есть некоторые тонкости, связанные с интернационализацией, так что в общем случае String.Compare правильнее. Но вы всё равно не знаете пока про Culture и тут строки сравниваются фактически как массивы байтов, так что `<` более чем пойдёт.